docs/external-module.format.md

```diff
@@ -0,0 +1,24 @@
+# External Module Format
+Typings are written in non-ambient external module format. Most of the time we refer it simply as external module format.
+
+This is the same format you will get if you run `tsc -d` without bundling the file (i.e., not using the `outFile` parameter).
+
+This document will gives you an overview of what is external module format and why the typings need to be written in this format.
+
+## What is ambient?
+Ambient is the term refers to somethings that is aware of or relate to its immediate surrounding. An ambient module is a module that awares how it fits in its surrounding.
```

Comment:
We have this in the main repo's README:

> For clarity about what ambient definitions are it's worth taking a look at the [TypeScript Handbook](http://www.typescriptlang.org/Handbook#modules-working-with-other-javascript-libraries).

It might be worth repeating that here.  I have to say I don't love this definition but I can't say I've yet seen a concise definition of "Ambient" either.  The useful thing about the TypeScript Handbook is that it has an example of both Ambient Internal Modules and Ambient External Modules.  Those examples are tremendously useful to grasping just what we're talking about.